### DISCUSSION TITLE
Merge queue

### DISCUSSION THREAD

**Discussion by nojaf** [dotnet/fsharp#14720] [Ideas]
I noticed that [Pull request merge queue](https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta/) is now in public beta.

Would there be any interest in this? The reason I'm asking is that it can be a bit frustrating when your PR is green, has multiple approvals and some other PR got merged in before your own.
Then you need to merge/rebase the latest changes and wait again for CI to complete.

With a merge queue, you would at least know that your is on its way to get merged.

**Comment by vzarytovskii** [comment]
You don't have to merge branch back to yours, unless it's a release one and as long as there's no merge conflicts.

**Reply by nojaf** [reply]
There was a time when being on top of the latest main was a requirement.
PRs like:
- https://github.com/dotnet/fsharp/pull/14148
- https://github.com/dotnet/fsharp/pull/13847

Have a lot of `Merge branch 'main' into `.
That is now no longer the case?

**Reply by vzarytovskii** [reply]
> There was a time when being on top of the latest main was a requirement.
> PRs like:
> - https://github.com/dotnet/fsharp/pull/14148
> - https://github.com/dotnet/fsharp/pull/13847
> 
> Have a lot of `Merge branch 'main' into `.
> That is now no longer the case?

Yeah, I've enabled it for all branches, but then reconsider ed main, since there's a bunch of traffic there, and it's not very convenient.

It's disabled there for some time now.

**Reply by nojaf** [reply]
Alright, great to hear.
